docs/2.9. Usage - Fully Automated Testing of NMOS Controllers.md

```diff
@@ -0,0 +1,174 @@
+# Testing NMOS Controllers
+
+## Fully Automated NMOS Controller Tests
+
+The testing tool supports fully automated testing of NMOS Controllers by replacing the Testing Façade with your own application to receive and process the requests from the NMOS Testing Tool. A full example utilising Flask and Selenium can be found on the nmos-js repository under AutomatedTesting. 
+
+## Building your fully automated test interface
+
+The Testing Façade IP/Hostname and Port on the test suite should be set to the location of your application.
+POST requests are sent from the Test Suite to the API endpoint `{IP/hostname}:{Port}/x-nmos/testing-facade` 
+
+The JSON schema for the Q&A protocol is as follows:
```

Comment:
Hm, interesting... this is a YAML representation of a JSON Schema?

Reply:
Hmm yes, that should be a json tag not yaml I think.

Reply:
I suspect an actual JSON Schema would be even more verbose than what's here right now. I think JSON Schema is a good idea, but how about including them as standalone files in the repo, something like _testingfacade-question.json_ and _testingfacade-answer.json_.

Actually, would it be nice to provide RAML also, and follow the pattern of the NMOS API definitions, i.e. something like this (maybe under the _testingfacade_ directory?)...

* _APIs/_
  * _TestingFacade.raml_
  * _schemas/_
    * _testingfacade-post-request.json_
    * _testingacade-post-response.json_
* _examples/_
  * _testingfacade-post-request.json_
  * _testingfacade-post-200.json_

Reply:
Yes, you're right. @hrbriggs did cut down the schema as it was far too verbose for the docs, and I think that you're suggestion of separate files is the right approach. I agree that RAML etc. consistent with other API documentation is the best approach.